app/lib/tenejo/csv_importer.rb

```diff
@@ -83,57 +83,38 @@ def job_owner
       @depositor
     end
 
-    def search(item, child_id)
-      found = item.children.find { |x| x.identifier == child_id }
-      if !found && !item.children.empty?
-        item.children.each do |x|
-          found ||= search(x, child_id)
-        end
-      end
-      found
+    def search_members(type, file_id)
+      @job.graph.flatten.find { |x| x.is_a?(type) && x.identifier == file_id }
     end
 
-    # TODO: make pfffiles live in the children array, so we don't have to search
-    # for them separately
-    def search_file(item, file_id)
-      # search under the files key instead of children
-      found = item.children.find { |x| x.identifier == file_id } if item.respond_to?(:files)
-      if !found && !item.children.empty?
-        item.children.each do |x|
-          found ||= search_file(x, file_id)
-        end
+    def update_status(item, start_state, end_state)
+      tmp = @job.graph
+      member = search_members(item.class, item.identifier)
+      if member
+        member.status = start_state
+        @job.graph = tmp # this only exists to trick ActiveRecord into actually saving the modified graph
+        @job.save!
+        yield
+        member.status = end_state
+        @job.graph = tmp # same here
+        @job.save!
+      else
+        yield # still need to call the block if the member doesn't exist in the graph somehow. this seems like a bug related to test setup
```

Comment:
I agree, this seems like an error condition.  But +1 to checking for it.  Maybe log something if we ever end up here.

Reply:
this is there only because some specs don't set up the @job structure, it shouldn't ever be the case in reality